### PR TITLE
Never pass a NoOp value to an op

### PR DESCRIPTION
**Pull request by mattsoulanille**
The 'NoOp' op appears in the execution graph (although it shouldn't really), and its output value can sometimes be passed to downstream nodes. Temporarily fix this by filtering it out of the parameters in `getParamValue`.

TODO(mattSoulanille): Refactor the graph so NoOp and other control dependencies are only used for sorting and don't appear in the execution graph.

To see the logs from the Cloud Build CI, please join either our [discussion](https://groups.google.com/a/tensorflow.org/forum/#!forum/tfjs) or [announcement](https://groups.google.com/a/tensorflow.org/forum/#!forum/tfjs-announce) mailing list.

- - -
This change is [Reviewable](https://reviewable.io/reviews/tensorflow/tfjs/7460)

### CODE REVIEW

**Review by pyu10055 — APPROVED** [review]
> Reviewed 4 of 4 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/tensorflow/tfjs/7460)* status: :shipit: complete! 1 of 1 approvals obtained

**Review by Linchenn — APPROVED** [review]
LGTM

**Comment by chunnienc** [comment]
It looks like most of the usage of getting tensor list param value treat it as unknown size list, but some of it does extract positional tensors from the list [(Code Search, notice the search result only convers function calls in a single line, so the actual use cases may be more.)](https://source.corp.google.com/search?q=getParamValue.%2Bas%5CsTensor%5C%5B%5C%5D&ss=piper%2FGoogle%2FPiper:google3%2Fthird_party%2Fjavascript%2Ftfjs%2F) But I believe this PR won't affect those since they param tensors shouldn't come from NoOp node.

**Review by chunnienc — APPROVED** [review]

File: .bazelrc [hunk 1]
```diff
@@ -79,8 +79,8 @@ test:macos --define DISPLAY=true
 test:windows --define DISPLAY=true --//:headless=false
 
 # Enable debugging tests with --config=debug
-run:debug --test_arg=--node_options=--inspect-brk --test_output=streamed --test_strategy=exclusive --test_timeout=999999 --nocache_test_results
-test:debug --test_arg=--node_options=--inspect-brk --test_output=streamed --test_strategy=exclusive --test_timeout=999999 --nocache_test_results
+run:debug --test_arg=--node_options=--inspect-brk --test_output=streamed --test_strategy=exclusive --test_timeout=36000 --nocache_test_results
```
[the remark this answers is not in the document]
Comment (mattsoulanille) [reply .bazelrc:82]: Yes. Newer versions of Bazel have a limit to the test_timeout and throw an error if it's over 12 hours.

[the remark this answers is not in the document]
Comment (Linchenn) [reply .bazelrc:82]: Thanks!

**Comment by mattsoulanille** [comment]
> It looks like most of the usage of getting tensor list param value treat it as unknown size list, but some of it does extract positional tensors from the list [(Code Search, notice the search result only convers function calls in a single line, so the actual use cases may be more.)](https://source.corp.google.com/search?q=getParamValue.%2Bas%5CsTensor%5C%5B%5C%5D&ss=piper%2FGoogle%2FPiper:google3%2Fthird_party%2Fjavascript%2Ftfjs%2F) But I believe this PR won't affect those since param tensors shouldn't come from NoOp node.

Yeah, and that's why I would like to rewrite this solution later, since it doesn't guarantee the length of the returned list if there's a `NoOp` node as an input. This kind of error should really be caught in graph analysis instead of model execution. I agree that for correct graphs, we should never encounter this kind of parameter shortening, but I still don't really like it.

**Pull request merged**